### ISSUE TITLE
liquibase db dump not working

### ISSUE THREAD

**Issue by devinrsmith** [dropwizard/dropwizard#1497]
https://dropwizard.github.io/dropwizard/0.9.2/docs/manual/migrations.html

suggests that `java -jar hello-world.jar db dump helloworld.yml` will generate a liquibase schema. I've definitely got a non-empty db, configured like:

```
  database:
    driverClass: org.h2.Driver
    user: username
    password: super_secret
    url: jdbc:h2:./target/mydb
```

I've added the correct bundle, and the command executes but outputs an empty xml tree:

```
<?xml version="1.0" encoding="UTF-8"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd"/>
```

**Comment by patrox** [comment]
hi @devinrsmith, can you please provide more details like which Dropwizard version are you using?

I'm asking, as I've checked it myself - and it worked as expected in `1.0.0-rc2-SNAPSHOT`.

> I've definitely got a non-empty db

The mentioned db was created by `db migrate` or was it created without using liquibase ?
Did you try to run the `dropwizard-example` migration ?

Once you provide more info, I'll try to help you.

**Comment by patrox** [comment]
Here is the result of my `db dump`:

```
dropwizard/dropwizard-example $ java -jar target/dropwizard-example-1.0.0-rc2-SNAPSHOT.jar db dump example.yml 
```
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
    <changeSet author="patrox (generated)" id="1459323075021-1">
        <createTable catalogName="EXAMPLE" schemaName="PUBLIC" tableName="PEOPLE">
            <column autoIncrement="true" name="ID" type="BIGINT(19)">
                <constraints primaryKey="true" primaryKeyName="PK_PEOPLE"/>
            </column>
            <column name="FULLNAME" type="VARCHAR(255)">
                <constraints nullable="false"/>
            </column>
            <column name="JOBTITLE" type="VARCHAR(255)"/>
        </createTable>
    </changeSet>
</databaseChangeLog>
```

**Comment by patrox** [comment]
Last thing which comes to my mind without more details:

>     user: username
    password: super_secret
    url: jdbc:h2:./target/mydb

Are you sure that these are correct ?

**Comment by devinrsmith** [comment]
I was using 0.9.2. It was on an H2 database that wasn't created with liquibase. (It was created with JDBI DAO.)

When I create a simple database with liquibase, I'm successfully able to dump it successfully.

I went ahead and wrote migrations.xml by hand for my case. I'm assuming that my underlying database had complex constraints that liquibase wasn't able to figure out. 

I didn't save the files, so I'm not able to share the db files. I can spend some time regenerating a similar schema via JDBI to reproduce.

**Comment by patrox** [comment]
There is also an option to create a copy of existing H2 DB using `Script` tool, which is part of H2:

Quote from http://www.h2database.com/html/tutorial.html:

> `java org.h2.tools.Script -url jdbc:h2:~/test -user sa -script test.zip -options compression zip`

I suspect that this might be an issue with liquibase - as you wrote that it worked for simpler DB, but I will wait for you feedback, as I was not able to reproduce it. If you could send a failing schema, then this would be really helpful.

**Comment by nickbabcock** [comment]
Closing this, as it seems due to Liquidbase and not Dropwizard itself.

**Comment by fspinnenhirn** [comment]
FYI:
I've encountered the exact same issue as @devinrsmith using dropwizard 1.0.0 - it turns out the Liquibase data dump used the _default schema_, whereas my tables were created in a different schema.  
The fix for issue #584 lets you specify the catalog and/or schema for which to generate the database changelog via the CLI parameters `--catalog` and `--schema`  
The example from the dropwizard documentation, but with a custom schema would look as follows:
```java -jar hello-world.jar db dump --schema mySchema helloworld.yml```
